src/memory.rs

```diff
@@ -127,7 +127,102 @@ pub fn to_java_heap_value(
+            BinaryMultiple::Exbi => self.scale_to(BinaryMultiple::Gibi),
+        }
+    }
+
+    /// Scale down the unit by one order of magnitude, i.e. GB to MB.
+    fn scale_down_unit(&self) -> OperatorResult<Self> {
+        match self.unit {
+            BinaryMultiple::Kibi => Err(Error::CannotScaleDownMemoryUnit),
+            BinaryMultiple::Mebi => Ok(self.scale_to(BinaryMultiple::Kibi)),
+            BinaryMultiple::Gibi => Ok(self.scale_to(BinaryMultiple::Mebi)),
+            BinaryMultiple::Tebi => Ok(self.scale_to(BinaryMultiple::Gibi)),
+            BinaryMultiple::Pebi => Ok(self.scale_to(BinaryMultiple::Tebi)),
+            BinaryMultiple::Exbi => Ok(self.scale_to(BinaryMultiple::Pebi)),
+        }
+    }
+
+    /// Floors the value of this MemoryQuantity.
+    pub fn floor(&self) -> Self {
+        Self {
+            value: self.value.floor(),
+            unit: self.unit,
+        }
+    }
+
+    /// If the MemoryQuantity value is smaller than 1 (starts with a zero), convert it to a smaller
+    /// unit until the non fractional part of the value is not zero anymore.
+    /// This can fail if the quantity is smaller than 1kB.
+    fn ensure_no_zero(&self) -> OperatorResult<Self> {
+        if self.value < 1. {
+            self.scale_down_unit()?.ensure_no_zero()
+        } else {
+            Ok(*self)
+        }
+    }
+
+    /// Ensure that the value of this MemoryQuantity is a natural number (not a float).
+    /// This is done by picking smaller units until the fractional part is smaller than the tolerated
+    /// rounding loss, and then rounding down.
+    /// This can fail if the tolerated rounding loss is less than 1kB.
+    fn ensure_integer(&self, tolerated_rounding_loss: MemoryQuantity) -> OperatorResult<Self> {
+        let fraction_memory = MemoryQuantity {
+            value: self.value.fract(),
+            unit: self.unit,
+        };
+        if fraction_memory < tolerated_rounding_loss {
+            Ok(self.floor())
+        } else {
+            self.scale_down_unit()?
+                .ensure_integer(tolerated_rounding_loss)
+        }
+    }
+
+    /// Returns a value like '1355m' or '2g'. Always returns natural numbers with either 'k', 'm' or 'g',
+    /// even if the values is multiple Terabytes or more.
+    /// The original quantity may be rounded down to achive a compact, natural number representation.
+    /// This rounding may cause the quantity to shrink by up to 20MB.
+    /// Useful to set memory quantities as JVM paramters.
+    pub fn format_for_java(&self) -> OperatorResult<String> {
+        let m = self
+            .scale_to_at_most_gb() // Java Heap only supports specifying kb, mb or gb
+            .ensure_no_zero()? // We don't want 0.9 or 0.2
+            .ensure_integer(MemoryQuantity::from_mebi(20.))?; // Java only accepts integers not floats
+        Ok(format!("{}{}", m.value, m.unit.to_java_memory_unit()))
+    }
```

Comment:
Why 20MB?

Reply:
Well, the other java formatting function uses 0.2 arbitrarily. thats why it's with a "2". But other than that, I don't know

Reply:
But the 0.2 is the leftover we would not use for heap allocation from the overall resources? Thats different. Or do i get something wrong here?